src/ceph-detect-init/CMakeLists.txt

```diff
@@ -1,8 +1,14 @@
 set(CEPH_DETECT_INIT_VIRTUALENV ${CEPH_BUILD_VIRTUALENV}/ceph-detect-init-virtualenv)
 
+if(WITH_PYTHON3)
+  set(CEPH_DETECT_INIT_PYTHON_BINARY python3)
+else()
+  set(CEPH_DETECT_INIT_PYTHON_BINARY python2.7)
+endif()
+
 add_custom_target(ceph-detect-init
   COMMAND
-  ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=python2.7 ${CEPH_DETECT_INIT_VIRTUALENV} &&
+  ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=${CEPH_DETECT_INIT_PYTHON_BINARY} ${CEPH_DETECT_INIT_VIRTUALENV} &&
```

Comment:
how about just dropping the `--python=python2.7` option? i knew this is not quite right either. because we intend to test ceph-detect-init using the python runtime it will be working with in user's environment.

Reply:
Right, and the ceph "Mimic" will be running in environments where there is not python 2.7 at all, only python 3. Hence these efforts.